src/representation_theory/Rep.lean

```diff
@@ -63,12 +73,120 @@ by apply_instance
+  (of_Module_monoid_algebra.obj M : Type u) = restrict_scalars k (monoid_algebra k G) M := rfl
+
+lemma of_Module_monoid_algebra_obj_ρ (M : Module.{u} (monoid_algebra k G)) :
+  (of_Module_monoid_algebra.obj M).ρ = representation.of_module k G M := rfl
+
+/-- Auxilliary definition for `equivalence_Module_monoid_algebra`. -/
+def counit_iso_add_equiv {M : Module.{u} (monoid_algebra k G)} :
+  ((of_Module_monoid_algebra ⋙ to_Module_monoid_algebra).obj M) ≃+ M :=
+begin
+  dsimp [of_Module_monoid_algebra, to_Module_monoid_algebra],
+  refine (representation.of_module k G ↥M).as_module_equiv.trans (restrict_scalars.add_equiv _ _ _),
+end
+
+/-- Auxilliary definition for `equivalence_Module_monoid_algebra`. -/
+def unit_iso_add_equiv {V : Rep k G} :
+  V ≃+ ((to_Module_monoid_algebra ⋙ of_Module_monoid_algebra).obj V) :=
+begin
+  dsimp [of_Module_monoid_algebra, to_Module_monoid_algebra],
+  refine V.ρ.as_module_equiv.symm.trans _,
+  exact (restrict_scalars.add_equiv _ _ _).symm,
+end
+
+/-- Auxilliary definition for `equivalence_Module_monoid_algebra`. -/
+def counit_iso (M : Module.{u} (monoid_algebra k G)) :
+  (of_Module_monoid_algebra ⋙ to_Module_monoid_algebra).obj M ≅ M :=
+linear_equiv.to_Module_iso'
+{ map_smul' := λ r x, begin
+    dsimp [counit_iso_add_equiv],
+    simp,
+  end,
+  ..counit_iso_add_equiv, }
+
+lemma unit_iso_comm (V : Rep k G) (g : G) (x : V) :
+  unit_iso_add_equiv (((V.ρ) g).to_fun x) =
+    (((of_Module_monoid_algebra.obj (to_Module_monoid_algebra.obj V)).ρ) g).to_fun
+      (unit_iso_add_equiv x) :=
+begin
+  dsimp [unit_iso_add_equiv, of_Module_monoid_algebra, to_Module_monoid_algebra],
+  simp only [add_equiv.apply_eq_iff_eq, add_equiv.apply_symm_apply,
+    representation.as_module_equiv_symm_map_rho, representation.of_module_as_module_act],
+end
+
+/-- Auxilliary definition for `equivalence_Module_monoid_algebra`. -/
+def unit_iso (V : Rep k G) :
+  V ≅ ((to_Module_monoid_algebra ⋙ of_Module_monoid_algebra).obj V) :=
+Action.mk_iso (linear_equiv.to_Module_iso'
+{ map_smul' := λ r x, begin
+    dsimp [unit_iso_add_equiv],
+    simp only [representation.as_module_equiv_symm_map_smul,
+      restrict_scalars.add_equiv_symm_map_algebra_map_smul],
+  end,
+  ..unit_iso_add_equiv, })
+  (λ g, by { ext, apply unit_iso_comm, })
+
+/-- The categorical equivalence `Rep k G ≌ Module (monoid_algebra k G)`. -/
+def equivalence_Module_monoid_algebra : Rep k G ≌ Module.{u} (monoid_algebra k G) :=
+{ functor := to_Module_monoid_algebra,
+  inverse := of_Module_monoid_algebra,
+  unit_iso := nat_iso.of_components (λ V, unit_iso V) (by tidy),
+  counit_iso := nat_iso.of_components (λ M, counit_iso M) (by tidy), }
+
 -- Verify that the symmetric monoidal structure is available.
 example : symmetric_category (Rep k G) := by apply_instance
 example : monoidal_preadditive (Rep k G) := by apply_instance
 example : monoidal_linear k (Rep k G) := by apply_instance
```

Comment:
Organizationally, should these go up to just before the `noncomputable theory`?

Reply:
LGTM